### PR TITLE
Add org.salutatoi.Cagou

### PR DESCRIPTION
**Pull request by goffi-contrib**
Cagou is the desktop/mobiles frontend of Salut à Toi. Salut à Toi is an
advanced XMPP client offering features such as end-to-end encryption
(OTR + OMEMO), file sharing, remote controls, etc.

### CODE REVIEW

File: org.salutatoi.Cagou.json [hunk 1]
```diff
@@ -0,0 +1,1463 @@
+{
+    "app-id": "org.salutatoi.Cagou",
+    "default-branch": "0.7.0",
```
Comment (barthalion) [inline org.salutatoi.Cagou.json]: It should be unset or `"branch": "stable"`.

File: org.salutatoi.Cagou.json [hunk 2]
```diff
@@ -0,0 +1,1463 @@
...
+    "finish-args": [
+        "--socket=session-bus",
+        "--socket=x11",
+        "--share=ipc",
+        "--socket=wayland",
+        "--share=network",
+        "--filesystem=home"
+    ],
+    "modules": [
+        {
```
Comment (barthalion) [inline org.salutatoi.Cagou.json]: Python 2 is available in https://github.com/flathub/shared-modules/

File: org.salutatoi.Cagou.json [hunk 3]
```diff
@@ -0,0 +1,1463 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://www.python.org/ftp/python/2.7.16/Python-2.7.16.tar.xz",
+                    "sha256": "f222ef602647eecb6853681156d32de4450a2c39f4de93bd5b20235f2e660ed7"
+                }
+            ]
+        },
+        {
+            "name": "libxslt",
```
Comment (barthalion) [inline org.salutatoi.Cagou.json]: All modules likely need some cleanups to delete development headers (etc.).

File: org.salutatoi.Cagou.json [hunk 4]
```diff
@@ -0,0 +1,1463 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "ftp://xmlsoft.org/libxml2/libxslt-1.1.33.tar.gz",
+                    "sha256": "8e36605144409df979cab43d835002f63988f3dc94d5d3537c12796db90e38c8"
+                }
+            ]
+        },
+        {
+            "name": "setuptools",
```
Comment (barthalion) [inline org.salutatoi.Cagou.json]: Are all these modules shared between Primitivus and Jp? I would consider keeping them in separate file and included as git submodule to ease maintenance.

File: org.salutatoi.Cagou.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,58 @@
+<?xml version='1.0' encoding='utf-8'?>
```
Comment (barthalion) [inline org.salutatoi.Cagou.appdata.xml]: Desktop and appdata files ideally should be part of upstream repositories and installed by project buildsystem if possible – other distributions are also using them.

**Comment by barthalion** [comment]
bot, build org.salutatoi.Cagou

**Comment by goffi-contrib** [comment]
Hello, thank for your review. I'll address the comments individually and make an update as soon as possible (quite busy right now).

File: org.salutatoi.Cagou.json (hunk 1 above)
Comment (goffi-contrib) [reply org.salutatoi.Cagou.json]: oh right, I'll change that thanks.

File: org.salutatoi.Cagou.json (hunk 2 above)
Comment (goffi-contrib) [reply org.salutatoi.Cagou.json]: I have seen indeed, but after doing that one unfortunately. I'll update to use the official one.

File: org.salutatoi.Cagou.json (hunk 3 above)
Comment (goffi-contrib) [reply org.salutatoi.Cagou.json]: I'll try to review non python modules to see what I can remove.

File: org.salutatoi.Cagou.json (hunk 4 above)
Comment (goffi-contrib) [reply org.salutatoi.Cagou.json]: yes `Primitivus` and `jp` share most of the dependencies (there are some more for Primitivus, notably `urwid`). But maintenance is really easy on my side, I've made a script to generate the manifest with the right dependencies in the right order automatically, and it is using a template so I can tell the script to change anything specific. I can maintain things easily like this and using git submodules would in fact complicate my workflow.

File: org.salutatoi.Cagou.appdata.xml (hunk 5 above)
Comment (goffi-contrib) [reply org.salutatoi.Cagou.appdata.xml]: Indeed. Next version will be Python 3 only and I've done many changes already, I'll add those script upstream for the next version, but I rather keep them here for this 0.7.0 version.

File: org.salutatoi.Cagou.json (hunk 4 above)
Comment (barthalion) [reply org.salutatoi.Cagou.json]: Alright, you're the captain here!

**Comment by goffi-contrib** [comment]
bot, build org.salutatoi.Cagou

**Comment by goffi-contrib** [comment]
@barthalion following your review, I've removed the `branch`, used flathub's shared-modules for `python2`, and removed `libxslt` after realizing it was in freedesktop runtime. I've also updated appdata file with `content_rating` as it was making the first test build fail.

I'm not sure why build are failing on non x86_64 platforms, `Pillow` complains that `The headers or library files could not be found for zlib` but zlib should be in freedesktop SDK right?

Let me know if there is anything else to change, thanks you :)

**Comment by nedrichards** [comment]
bot, build org.salutatoi.Cagou

**Comment by LeandroStanger** [comment]
bot, build org.salutatoi.Cagou

**Pull request closed without merging**